k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -1914,7 +1916,7 @@ void processPendingMoveOrCopy(PendingMoveOrCopy command, Account account) throws
             }
             if (!isCopy && Expunge.EXPUNGE_IMMEDIATELY == account.getExpungePolicy()) {
                 Timber.i("processingPendingMoveOrCopy expunging folder %s:%s", account.getDescription(), srcFolder);
-                remoteSrcFolder.expunge();
+                remoteSrcFolder.expunge(localSrcFolder.getAllMessageUids(true));
```

Comment:
With my suggestions applied this should only pass the UIDs of the moved messages to `expungeUids()`.